ethcore/src/miner/banning_queue.rs

```diff
@@ -78,6 +79,7 @@ impl BanningTransactionQueue {
 	/// May reject transaction because of the banlist.
 	pub fn add_with_banlist<F, G>(
 		&mut self,
+		chain: Option<&MiningBlockChainClient>,
```

Comment:
done, but maybe you have a better name for this trait (currently it is `TransactionDetailsProvider`) :)